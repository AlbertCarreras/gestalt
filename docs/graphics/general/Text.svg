<svg width="325" height="180" viewBox="0 0 325 180" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="325" height="180" fill="#F6FDF5"/>
<rect x="52" y="62" width="114" height="8" rx="4" fill="#005C62"/>
<rect x="52" y="94" width="166" height="8" rx="4" fill="#005C62"/>
<rect x="52" y="78" width="220" height="8" rx="4" fill="#005C62"/>
<rect x="52" y="110" width="185" height="8" rx="4" fill="#005C62"/>
<path d="M0 0V-2H-2V0H0ZM325 0H327V-2H325V0ZM325 180V182H327V180H325ZM0 180H-2V182H0V180ZM0 2H325V-2H0V2ZM323 0V180H327V0H323ZM325 178H0V182H325V178ZM2 180V0H-2V180H2Z" fill="#767676"/>
</svg>
