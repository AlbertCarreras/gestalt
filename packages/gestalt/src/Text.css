.Text {
  composes: antialiased sansSerif from "./Typography.css";
}

/* Link management */

/* .standalone is so we can select base on class */

.standalone {
  color: inherit;
}

.noOutline {
  outline: none;
}

.hoverNoUnderline:hover,
.hoverNoUnderline:focus {
  text-decoration: none;
}

.hoverUnderline:hover,
.hoverUnderline:focus {
  text-decoration: underline;
}

.outlineFocus:focus {
  outline: 4px solid var(--color-border-focus);
  outline-offset: 0;
}

.outlineFocusVR:focus {
  border-radius: var(--rounding-100);
  outline: 2px solid var(--color-border-focus);
  outline-offset: 2px;
}

/* Link color management */

.inheritColor {
  color: inherit; /* Links should have the correct underline color */
}

.vrInheritColor {
  color: inherit;
}

.dark a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-dark);
}

.dark a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-dark);
}

.inverse a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-inverse);
}

.inverse a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-inverse);
}

.error a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-error);
}

.error a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-error);
}

.light a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-light);
}

.light a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-light);
}

.link,
.default a.standalone {
  color: var(--color-text-link);
}

.link a.vrInheritColor:hover,
.default a.standalone:hover {
  color: var(--sema-color-hover-text-link);
}

.link a.vrInheritColor:active,
.default a.standalone:active {
  color: var(--sema-color-pressed-text-link);
}

.recommendation a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-recommendation);
}

.recommendation a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-recommendation);
}

.shopping a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-link);
}

.shopping a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-link);
}

.subtle a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-subtle);
}

.subtle a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-subtle);
}

.success a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-success);
}

.success a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-success);
}

.warning a.vrInheritColor:hover {
  color: var(--sema-color-hover-text-warning);
}

.warning a.vrInheritColor:active {
  color: var(--sema-color-pressed-text-warning);
}

/* Decoration */

.underline {
  text-decoration: underline;
}

.noUnderline {
  text-decoration: none;
}

/* Color */

.dark {
  color: var(--color-text-dark);
}

.textColorOverrideDark * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-dark) !important;
}

.default {
  color: var(--color-text-default);
}

.disabled {
  color: var(--color-text-disabled);
}

.inverse {
  color: var(--color-text-inverse);
}

.error {
  color: var(--color-text-error);
}

/* Used for error messaging in Toast */

.textColorOverrideError * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-light) !important;
  /* stylelint-disable-next-line declaration-no-important */
  font-weight: var(--font-weight-semibold) !important;
}

.subtle {
  color: var(--color-text-subtle);
}

.warning {
  color: var(--color-text-warning);
}

.recommendation {
  color: var(--color-icon-recommendation);
}

.success {
  color: var(--color-text-success);
}

.shopping {
  color: var(--color-text-shopping);
}

.light {
  color: var(--color-text-light);
}

.textColorOverrideLight * {
  /* stylelint-disable-next-line declaration-no-important */
  color: var(--color-text-light) !important;
}
