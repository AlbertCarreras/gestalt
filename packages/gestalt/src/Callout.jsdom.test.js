// @flow strict
import { render } from '@testing-library/react';
import Callout from './Callout.js';
import Text from './Text.js';

describe('Callout', () => {
  it('renders error variant', () => {
    const { container } = render(
      <Callout
        message="Insert a clever error callout message here"
        iconAccessibilityLabel="error"
        type="error"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders warning variant', () => {
    const { container } = render(
      <Callout
        message="Insert a clever warning callout message here"
        iconAccessibilityLabel="warning"
        type="warning"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders info variant', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        type="info"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders message + title', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        type="info"
        title="A Title"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders message + title + primaryAction with href', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        type="info"
        title="A Title"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders message + title + primaryAction without href', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ label: 'Visit Pinterest', accessibilityLabel: '' }}
        type="info"
        title="A Title"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders message + title + primaryAction + secondaryAction', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        secondaryAction={{
          href: 'pinterest.com/help',
          label: 'Learn more',
          accessibilityLabel: 'Learn more: callout messages',
        }}
        type="info"
        title="A Title"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders message + title + primaryAction + dismissButton', () => {
    const { container } = render(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: () => {},
        }}
        type="info"
        title="A Title"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders rich text message', () => {
    const { container } = render(
      <Callout
        dismissButton={{
          accessibilityLabel: 'Dismiss this banner',
          onDismiss: () => {},
        }}
        iconAccessibilityLabel="Info"
        message={
          <Text inline>
            You have invited{' '}
            <Text inline weight="bold">
              Leaf Media Agency
            </Text>{' '}
            to your business hierarchy. Once they accept, you will be able to manage their business
            account.
          </Text>
        }
        primaryAction={{
          accessibilityLabel: 'Resend invite',
          label: 'Resend invite',
        }}
        secondaryAction={{
          accessibilityLabel: 'Cancel invite',
          label: 'Cancel invite',
        }}
        title="You've sent an invite"
        type="info"
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('handles onDismiss callback', () => {
    const mockOnDismiss = jest.fn<[], void>();
    const { getByLabelText } = render(
      <Callout
        message="Insert a clever error callout message here"
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: mockOnDismiss,
        }}
        iconAccessibilityLabel="error"
        type="error"
      />,
    );
    // eslint-disable-next-line testing-library/prefer-screen-queries -- Please fix the next time this file is touched!
    getByLabelText('Dismiss banner').click();
    expect(mockOnDismiss).toHaveBeenCalled();
  });
});
