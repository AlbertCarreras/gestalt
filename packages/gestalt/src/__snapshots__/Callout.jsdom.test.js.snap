// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout renders error variant 1`] = `
<div>
  <div
    class="box errorWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="error"
            class="icon errorIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever error callout message here
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders info variant 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders message + title + primaryAction + dismissButton 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                A Title
              </div>
            </div>
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
      <div
        class="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
      >
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <a
            aria-label=""
            class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
            href="pinterest.com"
            rel=""
            tabindex="0"
          >
            <div
              class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
            >
              <div
                class="FlexItem"
              >
                <div
                  class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
                >
                  Visit Pinterest
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="rtlPos"
    >
      <button
        aria-label="Dismiss banner"
        class="parentButton"
        tabindex="0"
        type="button"
      >
        <div
          class="button tapTransition enabled"
        >
          <div
            class="pog transparent"
            style="height: 48px; width: 48px;"
          >
            <svg
              aria-hidden="true"
              aria-label=""
              class="icon defaultIcon iconBlock"
              height="16"
              role="img"
              viewBox="0 0 24 24"
              width="16"
            >
              <path
                d="test-file-stub"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Callout renders message + title + primaryAction + secondaryAction 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                A Title
              </div>
            </div>
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
      <div
        class="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
      >
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <a
            aria-label="Learn more: callout messages"
            class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg transparent"
            href="pinterest.com/help"
            rel=""
            tabindex="0"
          >
            <div
              class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
            >
              <div
                class="FlexItem"
              >
                <div
                  class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
                >
                  Learn more
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <a
            aria-label=""
            class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
            href="pinterest.com"
            rel=""
            tabindex="0"
          >
            <div
              class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
            >
              <div
                class="FlexItem"
              >
                <div
                  class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
                >
                  Visit Pinterest
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders message + title + primaryAction with href 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                A Title
              </div>
            </div>
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
      <div
        class="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
      >
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <a
            aria-label=""
            class="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
            href="pinterest.com"
            rel=""
            tabindex="0"
          >
            <div
              class="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
            >
              <div
                class="FlexItem"
              >
                <div
                  class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
                >
                  Visit Pinterest
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders message + title + primaryAction without href 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                A Title
              </div>
            </div>
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
      <div
        class="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
      >
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <button
            aria-label=""
            class="button hideOutline block accessibilityOutline parentButton"
            tabindex="0"
            type="button"
          >
            <div
              class="button hideOutline block accessibilityOutline tapTransition lg white enabled childrenDiv"
            >
              <div
                class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders message + title 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                A Title
              </div>
            </div>
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever info callout message here
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Callout renders rich text message 1`] = `
<div>
  <div
    class="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="Info"
            class="icon infoIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="box marginBottom2"
            >
              <div
                class="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
              >
                You've sent an invite
              </div>
            </div>
            <span
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              You have invited
               
              <span
                class="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold"
              >
                Leaf Media Agency
              </span>
               
              to your business hierarchy. Once they accept, you will be able to manage their business account.
            </span>
          </div>
        </div>
      </div>
      <div
        class="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
      >
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <button
            aria-label="Cancel invite"
            class="button hideOutline block accessibilityOutline parentButton"
            tabindex="0"
            type="button"
          >
            <div
              class="button hideOutline block accessibilityOutline tapTransition lg transparent enabled childrenDiv"
            >
              <div
                class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Cancel invite
              </div>
            </div>
          </button>
        </div>
        <div
          class="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
        >
          <button
            aria-label="Resend invite"
            class="button hideOutline block accessibilityOutline parentButton"
            tabindex="0"
            type="button"
          >
            <div
              class="button hideOutline block accessibilityOutline tapTransition lg white enabled childrenDiv"
            >
              <div
                class="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Resend invite
              </div>
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="rtlPos"
    >
      <button
        aria-label="Dismiss this banner"
        class="parentButton"
        tabindex="0"
        type="button"
      >
        <div
          class="button tapTransition enabled"
        >
          <div
            class="pog transparent"
            style="height: 48px; width: 48px;"
          >
            <svg
              aria-hidden="true"
              aria-label=""
              class="icon defaultIcon iconBlock"
              height="16"
              role="img"
              viewBox="0 0 24 24"
              width="16"
            >
              <path
                d="test-file-stub"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Callout renders warning variant 1`] = `
<div>
  <div
    class="box paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 warningWeak xsDirectionColumn xsDisplayFlex"
  >
    <div
      class="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
      style="width: 100%;"
    >
      <div
        class="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
      >
        <div
          class="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
        >
          <svg
            aria-label="warning"
            class="icon warningIcon iconBlock"
            height="32"
            role="img"
            viewBox="0 0 24 24"
            width="32"
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
        <div
          class="box marginBottomAuto marginTopAuto paddingX6"
          style="max-width: 648px;"
        >
          <div
            class="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
          >
            <div
              class="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
            >
              Insert a clever warning callout message here
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
